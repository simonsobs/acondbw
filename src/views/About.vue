<template>
  <v-container fill-height fluid>
    <v-row align="center" justify="center">
      <div>
        <div class="display-2 text-center">This is an about page</div>
        <div class="mt-5 text-center">
          <span class="font-italic font-weight-medium">Server version:</span>
          {{ version }}
        </div>
      </div>
    </v-row>
  </v-container>
</template>

<script>
import VERSION from "@/graphql/queries/Version.gql";

export default {
  data() {
    return {
      version: null
    };
  },
  apollo: {
    version: VERSION
  }
};
</script>
