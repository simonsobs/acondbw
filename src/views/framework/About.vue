<template>
  <v-container fill-height fluid>
    <v-row align="center" justify="center">
      <div>
        <div class="text-h3 text-center">This is an about page</div>
        <div class="mt-5 text-center">
          <span class="font-italic font-weight-medium">Server version:</span>
          {{ version }}
        </div>
      </div>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import { computed } from "vue";
import { useVersionQuery } from "@/graphql/codegen/generated";

const { data } = useVersionQuery();
const version = computed(() => data?.value?.version || "unknown");
</script>
