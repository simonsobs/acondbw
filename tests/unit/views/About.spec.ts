import Vue from "vue";
import Vuetify from "vuetify";
import { mount, createLocalVue } from "@vue/test-utils";

import About from "@/views/framework/About.vue";

Vue.use(Vuetify);

describe("About.vue", () => {
  let localVue: ReturnType<typeof createLocalVue>;
  let wrapper: ReturnType<typeof mount>;

  beforeEach(() => {
    localVue = createLocalVue();
    wrapper = mount(About, {
      localVue,
    });
    wrapper.setData({
      version: "0.0.1.test",
    });
  });

  it("test text", () => {
    const text = wrapper
      .text()
      .trim()
      .replace(/[\n\r]+/g, " ") // remove line breaks
      .replace(/ +/g, " "); // remove multiple consecutive spaces
    expect(text).toBe("This is an about page Server version: 0.0.1.test");
  });

  it("match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
