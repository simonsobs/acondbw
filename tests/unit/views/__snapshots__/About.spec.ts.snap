// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue match snapshot 1`] = `
<div class="container container--fluid fill-height">
  <div class="row align-center justify-center">
    <div>
      <div class="text-h3 text-center">This is an about page</div>
      <div class="mt-5 text-center"><span class="font-italic font-weight-medium">Server version:</span>
        0.0.1.test
      </div>
    </div>
  </div>
</div>
`;
